frontend/expr_inferrer.py

```diff
@@ -375,33 +376,74 @@ def _get_args_types(args, context, instance, solver):
     return args_types
 
 
-def _infer_annotated_function_call(args_types, solver, annotations):
-    result_type = solver.new_z3_const("call")
-    solver.annotation_resolver.add_annotated_function_axioms(args_types, solver, annotations, result_type)
+def _infer_annotated_function_call(args_types, solver, annotations, result_type):
+    return solver.annotation_resolver.get_annotated_function_axioms(args_types, solver, annotations, result_type)
 
-    return result_type
+
+def _infer_builtin_method(args_types, solver, context, result_type, method_name):
+    """Get the axioms of built-in methods inference"""
+    possible_methods = context.get_matching_methods(method_name)
+    method_axioms = []
+    for method in possible_methods:
+        cur_method_axioms = _infer_annotated_function_call(args_types, solver, method, result_type)
+        if cur_method_axioms is not None:
+            method_axioms.append(cur_method_axioms)
+    return method_axioms
 
 
 def infer_func_call(node, context, solver):
     """Infer the type of a function call, and unify the call types with the function parameters"""
     instance = None
+    result_type = solver.new_z3_const("call")
+    call_axioms = []
     if isinstance(node.func, ast.Attribute):
-        called, instance = infer(node.func, context, solver, True)
+        # Add axioms for built-in methods
+        instance = infer(node.func.value, context, solver)
         args_types = _get_args_types(node.args, context, instance, solver)
+        call_axioms += _infer_builtin_method(args_types, solver, context, result_type, node.func.attr)
     else:
         args_types = _get_args_types(node.args, context, instance, solver)
         if isinstance(node.func, ast.Name) and context.has_annotated_func(node.func.id):
-            return _infer_annotated_function_call(args_types, solver, context.get_annotated_func(node.func.id))
+            # Add axioms for built-in functions
+            annotated_func = context.get_annotated_func(node.func.id)
+            func_axioms = _infer_annotated_function_call(args_types, solver, annotated_func, result_type)
+            if func_axioms is not None:
+                call_axioms.append(func_axioms)
+
+    try:
+        # Add normal call inference axioms
         called = infer(node.func, context, solver)
-
-    result_type = solver.new_z3_const("call")
-
-    solver.add(axioms.call(called, args_types, result_type, solver.z3_types),
+        call_axioms += axioms.call(called, args_types, result_type, solver.z3_types)
+    except NameError as e:
+        # function name wasn't found in the context
+        if not call_axioms:
+            # no match for the function name in the context or in the built-ins
+            raise e
```

Comment:
Well yea I agree with you.

We no longer need this error for now after introducing the recently merged changes for annotated functions.

Such kind of error handling needs to be done in many other places in the project, so I think it's better to do it separately (not the scope of this PR).